### PR TITLE
client: simulate drawing from a deck

### PR DESCRIPTION
**Pull request by blin00**
...instead of sampling from a fixed distribution

### CODE REVIEW

File: client/src/gameLogic.tsx [hunk 1]
```diff
@@ -86,37 +144,43 @@ export function setPlayerWord(room: StartingPhase, playerName: string, word: str
 
 // Move a room from the StartingPhase to the HintingPhase.
 export function startGameRoom(room: StartingPhase): HintingPhase {
-    return {
+    let players = room.players.map((player, index) => {
```
Comment (goffrie) [inline client/src/gameLogic.tsx]: `const`, also below

File: client/src/gameLogic.tsx [hunk 2]
```diff
@@ -86,37 +144,43 @@ export function setPlayerWord(room: StartingPhase, playerName: string, word: str
...
+            },
+            hintsGiven: 0,
+        };
+    });
+
+    let deck = createShuffledDeck(flatten(players.map((player) => player.hand.letters)));
+    let dummies = dummyLettersForFreeHint(players.length).map((untilFreeHint) => ({
+        currentLetter: '#',    // placeholder to be replaced by cycleDummies
+        untilFreeHint: untilFreeHint + 1,
+    }));
```
Comment (goffrie) [inline client/src/gameLogic.tsx]: this is jank

File: client/src/gameLogic.tsx (hunk 1 above)
Comment (blin00) [reply client/src/gameLogic.tsx]: boom

File: client/src/gameLogic.tsx (hunk 2 above)
Comment (blin00) [reply client/src/gameLogic.tsx]: rip i wanted to reuse the draw dummies logic which is itself big jank

**Pull request merged**